### DISCUSSION TITLE
__FLAREACT_DATA and props loaded with every request

### DISCUSSION THREAD

**Discussion by traffisco** [flareact/flareact#80] [Q&A]
Hi,
Amazing project, this is exactly what I been looking for to power my project.

With every page load, I see the  __FLAREACT_DATA objects in the page source and the props loaded in the network tab.
Looking and the page source, it looks like it already fully rendered, 
If the page is already rendered and cached on the edge, why the unnecessary content and requests?

Thanks,
Vadim

**Comment by jplhomer** [comment]
Hi Vadim! Thanks for the question.

- The `__FLAREACT_DATA` object you see is for **initial page load only**.
- The initial page load should *not* be loading any additional props over AJAX
- Additional page navigation will load any page component JS as script tags and props over AJAX

Both the page HTML and props JSON are rendered and cached at the edge. If you see behavior counter to this, please open an issue and share an example!

**Comment by traffisco** [comment]
Thanks for the prompt response.
I'm looking at https://flareact.com/, can see `__FLAREACT_DATA` object and props in the network tab.

**Comment by jplhomer** [comment]
Ahh right, I see now. The props you're seeing are being fetched as part of a `<Link>` behavior: If Flareact sees a link component within the current viewport, it will prefetch the props to make navigation a little more quick.

This is the current behavior for [Next.js as well](https://nextjs.org/docs/getting-started) — but I agree, it feels a little counter-intuitive. Seems like a fun optimization we could add to `flareact/link` to not prefetch props if the current page path matches the link href.

**Comment by traffisco** [comment]
I tested with a simple page with some props fetched from remote(no links).  I see all the props in the __FLAREACT_DATA  object after the initial page load. Is this expected behavior?

**Comment by traffisco** [comment]
@jplhomer 
Hi again,

When a client lands on a route, if the route is cached, he served with the cached html otherwise, props are fetched, the html is rendered on the edge and served to the client(with the props in the script...).

The issue I'm having is with the prefetch flow. 
If the page contains `<Link>`  , the props for that route are fetched and if the user clicks the link, the app makes "fake navigation" using those props(very buggy atm).  This is fine for that specific client but if another client lands on that same page, the cache is not primed at cloudflare and the html will be rendered from scratch. 

Wouldn't it be better to do a hard prefetch(fetching the route from the worker, html will be rendered and cache primed)?
Then, when the user clicks the link, you can perform hard navigation to already primed route? 
All that `__FLAREACT_DATA` spam could be removed completely. This would be great for SEO, faster load times for the crawlers.

Did I get that flow right or I wasting our time?
Vadim

**Reply by jplhomer** [reply]
Hey Vadim! I don't really follow your thoughts above. I'd encourage you to check out the code powering page rendering and routing. If you have specific questions or implementation ideas beyond that point, feel free to open a PR.
